### PR TITLE
CB-9480 Authorize POST list endpoints

### PR DESCRIPTION
**Pull request by foldik**
Real UMS test:
http://ci-cloudbreak.eng.hortonworks.com/job/cloudbreak-real-ums-test/2021/

### CODE REVIEW

**Comment by horadla23** [comment]
before any review please test this with real-ums-test job

**Comment by foldik** [comment]
Real UMS test:
http://ci-cloudbreak.eng.hortonworks.com/job/cloudbreak-real-ums-test/2010/

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/model/And.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+
+public class And implements Authorization {
```
Comment (horadla23) [inline authorization-common/src/main/java/com/sequenceiq/authorization/service/model/And.java:16]: Maybe you should consider to rename these classes. They make sense for devs who is in the development of this feature, but other devs won't understand it at first. These names `And`, `Or`, etc. are very general. The same applies for `Authorization` interface.

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.sequenceiq.cloudbreak.domain.projection;
+
+public interface StackCrnView {
```
Comment (horadla23) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java:3]: I am not sure if this is needed, we have StackView and StackApiView also.

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java [hunk 3]
```diff
@@ -198,6 +199,12 @@
     @Query("SELECT s FROM Stack s WHERE s.resourceCrn = :crn")
     Optional<Stack> findByResourceCrn(@Param("crn") String crn);
 
+    @Query("SELECT s.environmentCrn FROM Stack s WHERE s.resourceCrn = :crn")
+    Optional<String> findEnvCrnByResourceCrn(@Param("crn") String crn);
+
+    @Query("SELECT s.resourceCrn as resourceCrn, s.environmentCrn as environmentCrn FROM Stack s WHERE s.resourceCrn IN (:crns)")
+    List<StackCrnView> findAllByResourceCrn(@Param("crns") Set<String> crn);
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java:206]: same here, maybe we should move this into StackViewRepository

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceBasedCrnProvider.java [hunk 4]
```diff
@@ -25,5 +27,9 @@ default String getResourceCrnByResourceName(String resourceName) {
         return Optional.empty();
     }
 
+    default Map<String, Optional<String>> getWithEnvironmentCrnsByResourceCrns(Collection<String> resourceCrns) {
```
Comment (horadla23) [inline authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceBasedCrnProvider.java:30]: 'with' not needed in the method name i guess

**Review by horadla23 — CHANGES_REQUESTED** [review]

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java (hunk 2 above)
Comment (foldik) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java:3]: The intention was to return the least amount of data from the database. Those views contain way more than we need here.

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceBasedCrnProvider.java (hunk 4 above)
Comment (foldik) [reply authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceBasedCrnProvider.java:30]: thanks

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java (hunk 2 above)
Comment (horadla23) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java:3]: It is clear, I am just not sure if there is significant difference between performances of querying these views. If it is, then it is reasonable.

Comment (foldik) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java:3]: I think this interface can be easily refactored, deleted if we would need more data for this feature.

Comment (horadla23) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/projection/StackCrnView.java:3]: alright

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/model/And.java (hunk 1 above)
Comment (foldik) [reply authorization-common/src/main/java/com/sequenceiq/authorization/service/model/And.java:16]: Okay I will rename those to `AuthorizationRule`, `AllMatch` and `AnyMatch`, although I would not give them thirty letters long names just to make it clear (actually more frustrating to read) every time, in every cases that this is about authorization.

**Review by lnardai — APPROVED** [review]

**Review by horadla23 — APPROVED** [review]

**Pull request merged**